src/Smalot/PdfParser/Encoding.php

```diff
@@ -145,6 +145,12 @@ protected function getEncodingClass(): string
     {
         // Load reference table charset.
         $baseEncoding = preg_replace('/[^A-Z0-9]/is', '', $this->get('BaseEncoding')->getContent());
+
+        // Check for empty BaseEncoding field value
+        if ('' == $baseEncoding) {
```

Comment:
Based on https://www.php.net/manual/en/function.preg-replace.php, `preg_replace` can return `string`, `array` and `null`. Shouldn't we check for `[]` and `null` at this point too?

Reply:
I could perhaps change the test to `if (empty($baseEncoding)) { ... }` ? How's that?

Reply:
I would change the whole part to:

```php
// Check for empty BaseEncoding field value
if (is_string($baseEncoding) && 0 < strlen($baseEncoding)) {
    $className = '\\Smalot\\PdfParser\\Encoding\\'.$baseEncoding;
} else {
    $baseEncoding = 'StandardEncoding';
}
```
Now we dont have to worry about null and empty arrays. I don't like `empty` personally, because it does too much magic and one may not be aware of its side "effects" (e.g. no warning if a variable doesn't exist). With this approach we are very explicit what we want and in all other cases we use `StandardEncoding`. 

However we do this code part, we must handle the null- and empty array case.

Reply:
Since the `$className = ...` statement is right below this, it would be simpler to use:

```php
if (!is_string($baseEncoding) || 0 == strlen($baseEncoding)) {
  $baseEncoding = 'StandardEncoding';
}
```

Reply:
My bad, I forgot to adapt one line. In the following would be the whole function with my adaptions:

```php
protected function getEncodingClass(): string
{
    // Load reference table charset.
    $baseEncoding = preg_replace('/[^A-Z0-9]/is', '', $this->get('BaseEncoding')->getContent());

    // use base encoding class if a non-empty string is provided
    if (is_string($baseEncoding) && 0 < strlen($baseEncoding)) {
        $className = '\\Smalot\\PdfParser\\Encoding\\'.$baseEncoding;
    } else {
        $className = '\\Smalot\\PdfParser\\Encoding\\StandardEncoding';
    }

    if (!class_exists($className)) {
        throw new EncodingNotFoundException('Missing encoding data for: "'.$baseEncoding.'".');
    }

    return $className;
}
```

But your approach is fine too, mine was just a minor tweak. Feel free to use what you want.